src/cryptojwt/jwe/fernet.py

```diff
@@ -0,0 +1,51 @@
+import base64
+import os
+from typing import Optional
+from typing import Union
+
+from cryptography.fernet import Fernet
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+from cryptojwt.jwe import Encrypter
+from cryptojwt.utils import as_bytes
+
+DEFAULT_ITERATIONS = 390000
+
+
+class FernetEncrypter(Encrypter):
+    def __init__(
+        self,
+        password: str,
+        salt: Optional[bytes] = "",
+        hash_alg: Optional[str] = "SHA256",
+        digest_size: Optional[int] = 0,
+        iterations: Optional[int] = DEFAULT_ITERATIONS,
+    ):
+        Encrypter.__init__(self)
+        if not salt:
+            salt = os.urandom(16)
+        else:
+            salt = as_bytes(salt)
+
+        _alg = getattr(hashes, hash_alg)
+        # A bit special for SHAKE* and BLAKE* hashes
+        if hash_alg.startswith("SHAKE") or hash_alg.startswith("BLAKE"):
+            _algorithm = _alg(digest_size)
```

Comment:
What if digest size is 0 or None?

Perhaps `digest_size or DEFAULT_DIGEST_SIZE` and define the a default?

Reply:
There doesn't seem to be a simple default size.
0 is allowed by the SHAKE* algorithms.
The two BLAKE* algorithms have different digest_sizes.
Hence, no easy default.

Reply:
If the `digest_size` is required, we should raise a `ValueError` if not provided (when required).

Reply:
I think that's up to the underlying library.
Things may change and I don't want to have to track that.